ianalyzer_readers/readers/rdf.py

```diff
@@ -19,39 +21,55 @@ class RDFReader(Reader):
     see [rdflib parsers](https://rdflib.readthedocs.io/en/stable/plugin_parsers.html).
     '''
 
-    def source2dicts(self, source: Source) -> Iterable[Document]:
+    def sources(self, **kwargs):
+        ''' The RDFReader class overrides, atypically, the `sources` property
+        to coerce the output of this function to RDF subjects, rather than file names.
+        In order to adjust the strategy by which RDF files are located,
+        override the `file_selector` function.
         '''
-        Given a RDF source file, returns an iterable of extracted documents.
+        files = self.file_selector()
+        graph = Graph()
+        for input_file in files:
+            graph.parse(input_file)
+        self.graph = graph
+        for subject in self.document_subjects(graph):
+            yield subject
+    
+    def file_selector(self) -> Iterable:
+        ''' A function to locate potential RDF files
+        This function assumes all files are contained within one directory
+        Override this function for more complex directory structure
+        and/or if rules for file selection should be applied.
+        '''
+        allowed_file_extensions = ['ttl', 'xml', 'json', 'rdf']
+        files = []
+        for extension in allowed_file_extensions:
+            files.extend(glob(f'{self.data_directory}/*.{extension}'))
+        return files
```

Comment:
Other reader classes like XMLReader don't implement file selection, and leave that to the child class. It's a design decision whether or not readers should include "default behaviour" for this; there are good arguments for either side, I think.

However, whatever decision is made here, it makes sense to make it consistently across readers. I don't see a reason why the RDFReader should behave differently from, say, the XMLReader in this respect. If you want update the package so all readers should include a default assumption of file structure, I would be open to that, but it's a separate issue.